AutomatedTesting/Gem/PythonTests/atom_renderer/atom_hydra_scripts/hydra_AtomEditorComponents_DecalAdded.py

```diff
@@ -0,0 +1,143 @@
+    6) Test IsHidden.
+    7) Test IsVisible.
+    8) Set Material property on Decal component.
+    9) Delete Decal entity.
+    10) UNDO deletion.
+    11) REDO deletion.
+    12) Look for errors.
+
+    :return: None
+    """
+    import os
+
+    import azlmbr.asset as asset
+    import azlmbr.bus as bus
+    import azlmbr.editor as editor
+    import azlmbr.legacy.general as general
+    import azlmbr.math as math
+
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    from editor_python_test_tools.utils import Report, Tracer, TestHelper as helper
+
+    with Tracer() as error_tracer:
+        # Test setup begins.
+        # Setup: Wait for Editor idle loop before executing Python hydra scripts then open "Base" level.
+        helper.init_idle()
+        helper.open_level("Physics", "Base")
+
+        # Test steps begin.
+        # 1. Create a Decal entity with no components.
+        decal_name = "Decal (Atom)"
+        decal_entity = EditorEntity.create_editor_entity_at(math.Vector3(512.0, 512.0, 34.0), f"{decal_name}")
+        Report.critical_result(Tests.decal_creation, decal_entity.exists())
+
+        # 2. Add Decal component to Decal entity.
+        decal_component = decal_entity.add_component(decal_name)
+        Report.critical_result(Tests.decal_component, decal_entity.has_component(decal_name))
+
+        # 3. UNDO the entity creation and component addition.
+        # Requires 3 UNDO calls to remove the Entity completely.
+        for x in range(4):
+            general.undo()
+        general.idle_wait_frames(1)
+        Report.result(Tests.creation_undo, not decal_entity.exists())
+
+        # 4. REDO the entity creation and component addition.
+        # Requires 3 REDO calls to match the previous 3 UNDO calls.
+        for x in range(4):
+            general.redo()
+        general.idle_wait_frames(1)
+        Report.result(Tests.creation_redo, decal_entity.exists())
+
+        # 5. Enter/Exit game mode.
+        helper.enter_game_mode(Tests.enter_game_mode)
+        general.idle_wait_frames(1)
+        helper.exit_game_mode(Tests.exit_game_mode)
+
+        # 6. Test IsHidden.
+        decal_entity.set_visibility_state(False)
+        is_hidden = editor.EditorEntityInfoRequestBus(bus.Event, 'IsHidden', decal_entity.id)
+        Report.result(Tests.is_hidden, is_hidden is True)
+
+        # 7. Test IsVisible.
+        decal_entity.set_visibility_state(True)
+        is_visible = editor.EditorEntityInfoRequestBus(bus.Event, 'IsVisible', decal_entity.id)
+        Report.result(Tests.is_visible, is_visible is True)
+
+        # 8. Set Material property on Decal component.
+        material_asset_path = os.path.join("AutomatedTesting", "Materials", "basic_grey.material")
+        decal_material_property_path = "Controller|Configuration|Material"
+        material_asset_id = asset.AssetCatalogRequestBus(
+            bus.Broadcast, "GetAssetIdByPath", material_asset_path, math.Uuid(), False)
```

Comment:
There is an Asset class utility that you can use as well

Reply:
So I was able to use this utility for the Global Skylight (IBL) component, but for the Decal component I got errors. I am able to use this current approach with the Decal component though, so keeping it for now.

I cut a ticket to figure out / investigate why the Asset class can't be used. Initial thoughts are that the current code is using a SOURCE image path over a Cache or "streamingimage" type file that the Global Skylight (IBL) component test has.